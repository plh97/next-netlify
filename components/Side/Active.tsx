"use client";
import React from "react";
import cs from "classnames";
import { usePathname } from "next/navigation";
import styles from "./active.module.scss";

interface IProps {
  path: string;
}

export default function Active({ path }: IProps) {
  const pathname = usePathname();
  let active = pathname.includes(path);
  if (path === "/") {
    active = pathname === path;
  }
  if (active) {
    return <span className={cs(styles.bg, { [styles.active]: active })} />;
  }
  return <span className={styles.bg} />;
}
