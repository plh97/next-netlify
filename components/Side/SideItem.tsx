"use client";
import Link from "next/link";
import cs from "classnames";
import { usePathname } from "next/navigation";
import React, { useCallback } from "react";

interface IProps {
  className: string;
  children: React.ReactNode;
  path: string;
  icon?: string;
}

export default function SideItem({ path, className, children, icon }: IProps) {
  const pathname = usePathname();
  const active = pathname === path;
  return (
    <Link href={path} className={cs(className, { active })}>
      <i className={cs("icon", icon)} />
      {children}
    </Link>
  );
}
