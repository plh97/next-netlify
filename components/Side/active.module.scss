.bg {
  width: 100%;
  height: 100%;
  position: absolute;
  left: 0;
  top: 0;
  &:hover {
    background-color: var(--grey8);
  }
}

.active {
  background-color: var(--grey8);
}