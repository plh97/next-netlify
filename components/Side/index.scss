.Side__container {
	background: var(--grey9);
	flex: 0 0 280px;
	display: flex;
	flex-direction: column;
	justify-content: space-between;
	font-size: 14px;

	.Side__nav {
		display: flex;
		flex-direction: column;
		padding-top: 20px;

		.Side__nav-item {
			margin-bottom: 10px;
			display: flex;
			flex-direction: row;
			align-items: center;
			justify-content: flex-start;
			height: 45px;
			padding: 0 20px;
			color: var(--grey1);
			transition: 0.2s all ease;
			box-sizing: border-box;
			border-left: 0px solid var(--blue1);

			.icon {
				margin-right: 20px;
				color: #fff;
				font-size: 18px;
			}

			&:hover,
			&.active {
				cursor: pointer;
				background: var(--grey8);

				a {
					color: var(--blue1);
				}
			}

			&.active {
				border-left: 3px solid var(--blue1);
			}
		}
	}

	.Side__history {
		flex: 0 0 300px;
		padding: 10px;

		.Side__history-title {
			font-size: 24px;
			color: #ccc;
		}

		.Side__history-content {
			line-height: 1.5em;
			padding-left: 20px;
			color: #999;

			a {
				margin-left: 10px;
				color: #999;
				text-decoration: none;

				&:hover {
					color: #00b4cf;
				}
			}
		}
	}
}
