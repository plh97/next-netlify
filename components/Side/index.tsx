import React from "react";
import "./index.scss";
import routes from "@/routes";

import SideItem from "./SideItem";

export default function Side() {
  return (
    <div className="Side__container">
      <div className="Side__nav">
        {routes
          .filter((route) => route.content)
          .map((route) => (
            <SideItem
              icon={route.icon}
              key={route.path}
              className="Side__nav-item"
              path={route.path ?? ""}
            >
              {route.content}
            </SideItem>
          ))}
      </div>
      {/* <div className="Side__history">
				<h3 className="Side__history-title">友链</h3>
				<ul className="Side__history-content">
					{[...new Array(10)].map((e, i) => (
						<li key={i}>
							<a href="void(0)">2e23e2e03/Acceptance_test</a>
						</li>
					))}
				</ul>
			</div> */}
    </div>
  );
}
