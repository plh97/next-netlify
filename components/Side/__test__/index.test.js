// Link.react.test.js
import React from 'react'
import Side from '../index.jsx'
import renderer from 'react-test-renderer'
import { MemoryRouter as Router } from 'react-router-dom' // 4.0.0

it('renders Side component correctly', () => {
	const tree = renderer
		.create(
			<Router>
				<Side />
			</Router>
		)
		.toJSON()
	expect(tree).toMatchSnapshot()
})
