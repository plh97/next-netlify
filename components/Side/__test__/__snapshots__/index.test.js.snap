// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Side component correctly 1`] = `
<div
  className="Side__container"
>
  <div
    className="Side__nav"
  >
    <a
      aria-current="page"
      className="Side__nav-item active"
      href="/"
      onClick={[Function]}
      style={Object {}}
    >
      <i
        className="icon icon-dashboard"
      />
      主页
    </a>
    <a
      aria-current={null}
      className="Side__nav-item"
      href="/article"
      onClick={[Function]}
    >
      <i
        className="icon icon-dashboard"
      />
      文章
    </a>
    <a
      aria-current={null}
      className="Side__nav-item"
      href="/repository"
      onClick={[Function]}
    >
      <i
        className="icon icon-dashboard"
      />
      仓库
    </a>
  </div>
</div>
`;
